### PR TITLE
Add pkg-config support to nxcomp, nxcompext and nxcompshad

### PR DESCRIPTION
**Pull request by uli42**
Add (rudimentary) pkg-config support to the nx libraries. 

The .pc files should also be included to the resulting -dev packages (for debian) but I do not know what needs to be done for that. So please can someone have a look a this?

### CODE REVIEW

**Comment by sunweaver** [comment]
@uli42: thanks for providing the .pc files.

File: nxcomp/Makefile.in [hunk 1]
```diff
@@ -58,14 +58,15 @@ LIBS        = @LIBS@
...
 #
 
-srcdir      = @srcdir@
-prefix      = @prefix@
-exec_prefix = @exec_prefix@
-bindir      = @bindir@
-man1dir     = @mandir@/man1
-VPATH       = @srcdir@
-libdir      = @libdir@
-includedir  = @includedir@
```
Comment (sunweaver) [inline nxcomp/Makefile.in:68]: unnecessary whitespace changes in this commit, please remove them from this commit.

File: nxcompext/Makefile.in [hunk 2]
```diff
@@ -50,14 +50,15 @@ LIBS        = @LIBS@ -lz -lNX_X11 -lXcomp
...
 #
 
-srcdir      = @srcdir@
-prefix      = @prefix@
-exec_prefix = @exec_prefix@
-bindir      = @bindir@
-man1dir     = @mandir@/man1
-VPATH       = @srcdir@
-libdir = @libdir@
-includedir = @includedir@
```
Comment (sunweaver) [inline nxcompext/Makefile.in:60]: again, unnecessary whitespace changes... please remove...

File: nxcompshad/Makefile.in [hunk 3]
```diff
@@ -68,14 +68,15 @@ LIBS        = @LIBS@ -lNX_X11
...
 #
 
-srcdir      = @srcdir@
-prefix      = @prefix@
-exec_prefix = @exec_prefix@
-bindir      = @bindir@
-man1dir     = @mandir@/man1
-VPATH       = @srcdir@
-libdir      = @libdir@
-includedir  = @includedir@
```
Comment (sunweaver) [inline nxcompshad/Makefile.in:78]: unnecessary whitespace changes (see above)

**Comment by sunweaver** [comment]
Please fix the non-needed whitespace changes in this PR (with commit --amend) and force-push the modified commit to the same PR branch.

The rest looks ok.

**Comment by uli42** [comment]
? If I skip the whitespace changes the files look garbled. I just adapted to the way the files are built already: all = chars in a section are placed at the same column.

**Comment by Ionic** [comment]
I agree with Uli on that. He had to add a single space char to match the equal signs, due to `pkgconfigdir` being longer than all other variable names.

LGTM, too.

**Comment by sunweaver** [comment]
> I agree with Uli on that. He had to add a single space char to match the
> equal sign up, due to `pkgconfigdir` being longer than all other
> variable names.

Yeah, I missed the necessity of inserting the one whitespace. It looked superfluous on the first spot.

> LGTM, too.

I will merge later today (or tomorrow)...
